lightspark-sdk/src/commonTest/kotlin/com/lightspark/sdk/ClientIntegrationTests.kt

```diff
@@ -309,6 +309,46 @@ class ClientIntegrationTests {
         payments[0].id.shouldBe(payment?.id)
     }
 
+    // todo add payment has tests
+
+    @Test
+    fun `test getOutgoingPaymentsForPaymentsHash`() = runTest {
+        val node = getFirstOskNode()
+        client.loadNodeSigningKey(node.id, PasswordRecoverySigningKeyLoader(node.id, NODE_PASSWORD))
+        val invoice = client.createTestModeInvoice(node.id, 100_000, "test invoice")
+        var outgoingPayment: OutgoingPayment? = client.payInvoice(node.id, invoice, maxFeesMsats = 100_000)
+        outgoingPayment.shouldNotBeNull()
+        while (outgoingPayment?.status == TransactionStatus.PENDING) {
+            delay(500)
+            outgoingPayment = OutgoingPayment.getOutgoingPaymentQuery(outgoingPayment.id).execute(client)
+            println("Payment status: ${outgoingPayment?.status}")
+        }
+        outgoingPayment?.status.shouldBe(TransactionStatus.SUCCESS)
+
+        val payments = client.getOutgoingPaymentsForPaymentsHash(outgoingPayment?.transactionHash!!)
```

Comment:
Does this work? I don't think the transactionHash is the right hash here. It should actually be the invoice's payment hash.

Reply:
They should be the same hash iiuc

Reply:
that being said, even if it's the same hash, for clarity it should probably come from the invoice payment hash?